tools/loader/ImageClassifier.cpp

```diff
@@ -430,25 +440,50 @@ int main(int argc, char **argv) {
   llvm::outs() << "Model: " << loader.getFunction()->getName() << "\n";
 
   int numErrors = 0;
+
+  // Image tensors loaded up to be run at once for benchmark mode.
+  std::vector<std::unique_ptr<Tensor>> preLoadedImages;
+  std::vector<std::unique_ptr<ExecutionContext>> contexts;
+
   while ((streamInputFilenamesMode &&
           getNextImageFilenames(&inputImageBatchFilenames)) ||
          (miniBatchMode &&
           getNextMiniBatch(inputImageBatchFilenames, inputImageFilenames,
                            minibatchIndex, miniBatch)) ||
          isFirstRun) {
-    // Load and process the image data into the inputImageData Tensor.
-    loadImagesAndPreprocess(inputImageBatchFilenames, &inputImageData,
-                            imageNormMode, imageChannelOrder, imageLayout);
-
+    auto inputImageData = llvm::make_unique<Tensor>();
```

Comment:
This is shadowing the `inputImageData` in the enclosing scope.  Is there any reason to move this inside the loop and unique-ptr it, rather than leaving it stack allocated outside the loop?  I strongly prefer the latter if possible.

Reply:
The unique pointer is because we plan on moving it into a vector and storing it, a single stack allocated image won't work for that.